suse_migration_services/fstab.py

```diff
@@ -114,4 +116,53 @@ def export(self, filename):
                 )
 
     def get_devices(self):
-        return self.fstab
+        return self._get_canonical_mount_list()
+
+    def _get_canonical_mount_list(self):
+        """
+        Implements hierarchical sorting of mount paths
+
+        :return: list of canonical fstab_entry_type elements
+
+        :rtype: list
+        """
+        mount_paths = {}
+        sorted_fstab = []
+        for entry in self.fstab:
+            mount_paths[entry.mountpoint] = entry
+        for mountpath in Fstab._sort_by_hierarchy(sorted(mount_paths.keys())):
+            sorted_fstab.append(mount_paths[mountpath])
+        return sorted_fstab
+
+    @staticmethod
+    def _sort_by_hierarchy(path_list):
+        """
+        Sort given list of path names by their hierachy in the tree
+
+        Example:
+
+        .. code:: python
+
+            result = sort_by_hierarchy(['/var/lib', '/var'])
```

Comment:
a bit nitpicking but it would be nice to have the call and the return value
```python
sort_by_hierarchy(['/var/lib', '/var'])
[output_sorted]
```

Reply:
Could you be more precise what exactly you expect in the Example code ? Please remember it has to be valid python otherwise the doc generator will fail. So currently we have this:

```
Example:

.. code:: python

     result = sort_by_hierarchy(['/var/lib', '/var'])
```

How would you like to change this to be more obvious ?

Thanks